src/main/java/practice/StreamPractice.java

```diff
@@ -14,28 +21,48 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
-    }
+        final String separator = ",";
 
+        OptionalInt minValue = numbers.stream()
+                .flatMap(s -> Arrays.stream(s.split(separator)))
+                .mapToInt(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
+                .min();
+        return minValue.orElseThrow(()
+                -> new RuntimeException("Can't get min value from list: "
+                + numbers));
+    }
     /**
      * Given a List of Integer numbers,
      * return the average of all odd numbers from the list or throw NoSuchElementException.
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
+
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        OptionalDouble optionalDouble
+                = IntStream.iterate(0, i -> i < numbers.size(), i -> i + 1)
+                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 != 0)
+                .average();
+        return optionalDouble.orElseThrow(NoSuchElementException::new);
```

Comment:
Here we can use just `getAsDouble()`. It itself throws `NoSuchElementException`.
```suggestion
                .average();
        return optionalDouble.getAsDouble();
```